Description: add sunway support
Author: Miao Changwei <miaochangwei@uniontech.com>

---
Last-Update: 2022-05-15

--- liburing-2.1.orig/src/syscall.c
+++ liburing-2.1/src/syscall.c
@@ -13,7 +13,7 @@
 
 #ifdef __alpha__
 /*
- * alpha and mips are exception, other architectures have
+ * alpha, mips and sunway are exception, other architectures have
  * common numbers for new system calls.
  */
 # ifndef __NR_io_uring_setup
@@ -35,7 +35,17 @@
 # ifndef __NR_io_uring_register
 #  define __NR_io_uring_register        (__NR_Linux + 427)
 # endif
-#else /* !__alpha__ and !__mips__ */
+#elif defined __sw_64__
+# ifndef __NR_io_uring_setup
+#  define __NR_io_uring_setup		272
+# endif
+# ifndef __NR_io_uring_enter
+#  define __NR_io_uring_enter		273
+# endif
+# ifndef __NR_io_uring_register
+#  define __NR_io_uring_register	274
+# endif
+#else /* !__alpha__ and !__mips__ and !__sw_64__ */
 # ifndef __NR_io_uring_setup
 #  define __NR_io_uring_setup		425
 # endif
